examples/simple-example.rs

```diff
@@ -126,46 +176,31 @@ impl<F: FieldExt> FieldChip<F> {
             s * (p + a * -F::one())
         });
 
-        FieldConfig {
+        FieldConfigEnum::Config(FieldConfig {
             advice,
             perm,
             s_mul,
             s_pub,
-        }
-    }
-}
-// ANCHOR_END: chip-config
-
-// ANCHOR: chip-impl
-impl<F: FieldExt> Chip for FieldChip<F> {
-    type Config = FieldConfig;
-    type Loaded = ();
-    type Field = F;
-
-    fn load(_layouter: &mut impl Layouter<Self>) -> Result<(), halo2::plonk::Error> {
-        // None of the instructions implemented by this chip have any fixed state.
-        // But if we required e.g. a lookup table, this is where we would load it.
-        Ok(())
+        })
     }
 }
 // ANCHOR_END: chip-impl
 
 // ANCHOR: instructions-impl
-/// A variable representing a number.
-#[derive(Clone)]
-struct Number<F: FieldExt> {
-    cell: Cell,
-    value: Option<F>,
-}
 
-impl<F: FieldExt> NumericInstructions for FieldChip<F> {
+impl<F: FieldExt> NumericInstructions<F> for FieldChip<F> {
     type Num = Number<F>;
 
     fn load_private(
-        layouter: &mut impl Layouter<Self>,
-        value: Option<Self::Field>,
+        &self,
+        mut layouter: impl Layouter<F>,
+        value: Option<F>,
     ) -> Result<Self::Num, Error> {
-        let config = layouter.config().clone();
+        let config = match self.config() {
+            FieldConfigEnum::Config(config) => config.clone(),
+            _ => unreachable!(),
+        };
```

Comment:
After the removal of `FieldConfigEnum`, this can just be:
```suggestion
        let config = self.config();
```
or
```suggestion
        let config = &self.config;
```
(and ditto below).